utils/python/CIME/buildlib.py

```diff
@@ -0,0 +1,49 @@
+from CIME.XML.standard_module_setup import *
+from CIME.case import Case
+"""
+buildlibx - given the caseroot and component name(cmp), build the stub component
+"""
+
+class buildlib(object):
+    def __init__(self, case, model, comp):
+        self.case = case
+        self.model = model
+        self.comp = comp
+        self.caseroot = case.get_value("CASEROOT")
+        self.casetools = case.get_value("CASETOOLS")
+        self.cimeroot = case.get_value("CIMEROOT")
+        self.mach = case.get_value("MACH")
+        self.objroot = case.get_value("OBJROOT")
+        self.libroot = case.get_value("LIBROOT")
+        self.gmake_j = case.get_value("GMAKE_J")
+        self.gmake = case.get_value("GMAKE")
+
+class buildxlib(buildlib):
+    def buildlib(self):
+        try:
```

Comment:
Standard safe way to open file:

    with open('Filepath', 'w') as out:
        * do stuff to out